### ISSUE TITLE
Investigate `object-literal-sort-keys`

### ISSUE THREAD

**Issue by kourge** [UrbanDoor/lint#9]
We currently disable [`object-literal-sort-keys`](https://palantir.github.io/tslint/rules/object-literal-sort-keys/). There is utility in this rule, but only more so when dealing with a large object literal. Ideally this should be determined by a configurable threshold, e.g. "enforce this when an object literal exceeds five entries."

Options bikeshedding, per my opinion:

- `ignore-case`: ambivalent
- `match-declaration-order`: this may be a great way to tide us over the lack of threshold. My initial objections against this rule is the fact most object literals have some kind of hidden logical order that is not alphabetic. This options instead enforces the literal to match the order of the properties as they are declared in the type, which solves this issue, assuming the type declaration's order is good.
- `shorthand-first`: ambivalent

**Comment by squirly** [comment]
I like the idea of having a threshold.
Is it possible to only do `match-declaration-order`, when it is available, otherwise preserve written order?

**Comment by kourge** [comment]
I don't think so. This rule [directly falls back to alphabetical when declaration order matching isn't possible](https://github.com/palantir/tslint/blob/master/src/rules/objectLiteralSortKeysRule.ts#L139), and there aren't other rules that remotely look like they only do declaration order.
